### DISCUSSION TITLE
Invoke-Command2 Using Port 0 on fresh install?

### DISCUSSION THREAD

**Discussion by malcolmcarey** [dataplat/dbatools#8500] [Q&A]
Wanted to see if anyone else has experienced this. I'm a new user here, never been a part of an open source project before and new to powershell.

I installed the module using Install-Module -Name dbatools

I then tried to run a command on a remote computer hosting a SQL Server instance, which is both on the same domain as me and I'm an administrator on the machine. See the results below (note I blocked out my server name and IP, the last line is really the important one).

`Get-DbaNetworkConfiguration -Sqlinstance 'myServer' -Verbose`

```
PS C:\WINDOWS\system32> Get-DbaNetworkConfiguration -Sqlinstance myServer -Verbose
VERBOSE: [18:15:04][Resolve-DbaNetworkName] Resolving myServer using .NET.Dns GetHostEntry
VERBOSE: [18:15:04][Resolve-DbaNetworkName] Resolving XXX using .NET.Dns GetHostByAddress
VERBOSE: [18:15:04][Get-DbaCmObject] Configuration loaded | Cache disabled: False
VERBOSE: [18:15:04][Get-DbaCmObject] [myServer] Retrieving Management Information
VERBOSE: [18:15:04][Get-DbaCmObject] [myServer] Accessing computer using Cim over WinRM
VERBOSE: [18:15:04][Get-DbaCmObject] [myServer] Accessing computer using Cim over WinRM - Success
VERBOSE: [18:15:04][Invoke-Command2] Using Port: 0
VERBOSE: [18:15:04][Resolve-DbaNetworkName] Unable to get DNS domain information from myServer
VERBOSE: [18:15:04][Resolve-DbaNetworkName] Resolving myServer using .NET.Dns GetHostEntry
VERBOSE: [18:15:04][Resolve-DbaComputerName] Resolved 'myServer' to 'myServer'
VERBOSE: [18:15:05][Invoke-Command2] Using Port: 0
WARNING: [18:15:05][Get-DbaNetworkConfiguration] Failed to collect network configuration from myServer for instance MSSQLSERVER. | The 0 argument is less than the minimum allowed range of 1. Supply an argument that is greater
than or equal to 1 and then try the command again.
```

So a port of 0 is being passed to the Invoke-Command2 function here.

In the code for Invoke-Command2, a port of 0 appears to be allowed. However, I don't really know how 0 is being passed as my port here, I would have assumed it would be $null. Alas, the error message indicates otherwise:

```
if (($null -ne $Port) -and ($Port -gt -1)) {
    $psSessionSplat.Port = $Port
    Write-Message -Level Verbose -Message "Using Port: $($psSessionSplat.Port)"
}
```

However, Invoke-Command requires the port number to be at least 1, at least on my machine. You can replicate that error message on your own machine just by trying to run Invoke-Command with -Port 0

I set my default port using:

`Set-DbatoolsConfig -FullName psremoting.pssession.port -Value 5985`

After that, running dbatools commands that use Invoke-Command 2 work just fine.

Not sure if this qualifies as a bug or not? I feel like it should, and I would vote for having the default psremoting.pssession.port be 5985, the same default port that is assigned when creating a new PSSession without specifying a port.

**Comment by ClaudioESSilva** [comment]
Hi @malcolmcarey,

Thanks for bringing this to our attention.

Yes, I see it as a bug. The idea is to keep the default behaviour of the command. So, on a new installation if you don't change the `psremoting.pssession.port` it should just work (because it will defaults on PSRemoting native commands default values)

Therefore I have opened the issue https://github.com/dataplat/dbatools/issues/8501 and submitted a PR with a fix https://github.com/dataplat/dbatools/pull/8502

Do you mind trying the new version once @potatoqualitee generates a new version?
